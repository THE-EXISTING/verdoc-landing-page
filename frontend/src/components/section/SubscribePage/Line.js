import React from 'react'
import styled from 'styled-components'
import media from 'styled-media-query'
import LineInteractive from '../../common/LineInteractive'
import R from '../../resources/R'

const WraperContent = styled.div`
  margin-top: 3.1rem;
  width: 100%;
`
const TextOnBtnFront = styled.div`
  font-size: 1em;
  color: ${R.colors.black};
  letter-spacing: 0.36px;
  margin: auto 14px 0 14px;

  ${media.lessThan('medium')`
        margin-left: 3em;
        margin-right: 3em;
    `}
`
const SetLowOpacity = styled.span`
  opacity: 0.6;
`
const LinkLine = styled.a`
  cursor: pointer;
  display: inline;
  font-weight: bold;
  color: ${R.colors.line};
  text-decoration: underline;
`
const TextRed = styled.span`
  font-weight: bold;
  color: ${R.colors.repeat};
`
const TextPurple = styled.span`
  font-weight: bold;
  color: ${R.colors.primary};
`

const Line = () => (
  <>
    <WraperContent>
      <TextOnBtnFront>
        <SetLowOpacity>Just scan</SetLowOpacity>
        <TextRed>!!</TextRed> <SetLowOpacity>on</SetLowOpacity>{' '}
        <LinkLine href="https://line.me/" target="_blank">
          Line
        </LinkLine>{' '}
        <SetLowOpacity> application and use </SetLowOpacity>
        <TextPurple>Verdoc.</TextPurple>
      </TextOnBtnFront>
      <LineInteractive />
    </WraperContent>
  </>
)
export default Line
