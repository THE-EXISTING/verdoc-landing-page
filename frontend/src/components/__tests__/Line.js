import React from 'react'
import renderer from 'react-test-renderer'

import Line from '../section/Line'

describe('Line', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Line />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
