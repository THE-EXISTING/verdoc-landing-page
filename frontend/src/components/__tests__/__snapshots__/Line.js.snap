// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Line renders correctly 1`] = `
<div
  className="sc-gqjmRU kehxiU"
>
  <div
    className="sc-VigVT gypDXo"
  >
    <span
      className="sc-jTzLTM gQYRF"
    >
      Just scan
    </span>
    <span
      className="sc-jzJRlG cJzsTM"
    >
      !!
    </span>
     
    <span
      className="sc-jTzLTM gQYRF"
    >
      on
    </span>
     
    <a
      className="sc-fjdhpX bZMazs"
      href="https://line.me/"
      target="_blank"
    >
      Line
    </a>
     
    <span
      className="sc-jTzLTM gQYRF"
    >
       application and use 
    </span>
    <span
      className="sc-cSHVUG jtAWWs"
    >
      Verdoc.
    </span>
  </div>
  <div
    className="sc-bxivhb fpYHvD"
  >
    <img
      className="sc-ifAKCX hDmxfN"
      onClick={[Function]}
      src="test-file-stub"
    />
    <p
      className="sc-EHOje gRMGc"
    >
      or
    </p>
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiButton-root sc-bZQynM hZkxnO MuiButton-outlined"
      href="line://ti/p/%40abr9961f"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      rel="noreferrer"
      role="button"
      tabIndex={0}
      target="_blank"
    >
      <span
        className="MuiButton-label"
      >
        <div
          className="sc-gzVnrw dDoANy"
        >
          <img
            className="sc-dnqmqq cUevqg"
            src="test-file-stub"
          />
          <div
            className="sc-htoDjs ufayu"
          >
            <img
              className="sc-iwsKbI dLTHSh"
              src="test-file-stub"
            />
            <div
              className="sc-gZMcBi jDZGjU"
            >
              LINE@
            </div>
          </div>
        </div>
      </span>
    </a>
  </div>
</div>
`;
